tests/mechanic/java_resolver_test.py

```diff
@@ -18,47 +18,35 @@
 import unittest.mock as mock
 from unittest import TestCase
 
-from esrally import config, exceptions
+from esrally import exceptions
 from esrally.mechanic import java_resolver
 
 
 class JavaResolverTests(TestCase):
     @mock.patch("esrally.utils.jvm.resolve_path")
     def test_resolves_java_home_for_default_runtime_jdk(self, resolve_jvm_path):
         resolve_jvm_path.return_value = (12, "/opt/jdk12")
-
-        cfg = config.Config()
-        cfg.add(config.Scope.application, "mechanic", "runtime.jdk", None)
-
         major, java_home = java_resolver.java_home("12,11,10,9,8",
-                                                   specified_runtime_jdk=cfg.opts("mechanic", "runtime.jdk"),
+                                                   specified_runtime_jdk=None,
                                                    provides_bundled_jdk=True)
 
-        self.assertEqual(major, 12)
-        self.assertEqual(java_home, "/opt/jdk12")
+        self.assertEqual(major, resolve_jvm_path.return_value[0])
+        self.assertEqual(java_home, resolve_jvm_path.return_value[1])
 
     @mock.patch("esrally.utils.jvm.resolve_path")
     def test_resolves_java_home_for_specific_runtime_jdk(self, resolve_jvm_path):
         resolve_jvm_path.return_value = (8, "/opt/jdk8")
-
-        cfg = config.Config()
-        cfg.add(config.Scope.application, "mechanic", "runtime.jdk", 8)
-
         major, java_home = java_resolver.java_home("12,11,10,9,8",
-                                                   specified_runtime_jdk=cfg.opts("mechanic", "runtime.jdk"),
+                                                   specified_runtime_jdk=8,
                                                    provides_bundled_jdk=True)
 
-        self.assertEqual(major, 8)
-        self.assertEqual(java_home, "/opt/jdk8")
+        self.assertEqual(major, resolve_jvm_path.return_value[0])
```

Comment:
I think the previous implementation was more readable, even if we duplicate code. However, the scope of duplication is within the test method so this is ok IMHO.